catalogue_pipeline/reindexer_v2/reindex_shard_generator/src/test_reindex_shard_generator.py

```diff
@@ -0,0 +1,195 @@
+# -*- encoding: utf-8 -*-
+
+import json
+import os
+
+import pytest
+
+from reindex_shard_generator import main
+
+
+@pytest.fixture(scope='session')
+def table_name():
+    name = 'test_reindex_shard_generator-table'
+    os.environ.update({'TABLE_NAME': name})
+    return name
+
+
+@pytest.fixture(scope='session')
+def source_data_table(dynamodb_client, table_name):
+    dynamodb_client.create_table(
+        TableName=table_name,
+        KeySchema=[
+            {
+                'AttributeName': 'id',
+                'KeyType': 'HASH',
+            }
+        ],
+        AttributeDefinitions=[
+            {
+                'AttributeName': 'id',
+                'AttributeType': 'S',
+            }
+        ],
+        ProvisionedThroughput={
+            'ReadCapacityUnits': 1,
+            'WriteCapacityUnits': 1,
+        }
+    )
+
+    # Quoting from the boto3 docs:
+    #
+    #   CreateTable is an asynchronous operation. Upon receiving a CreateTable
+    #   request, DynamoDB immediately returns a response with a TableStatus
+    #   of CREATING . After the table is created, DynamoDB sets the
+    #   TableStatus to ACTIVE . You can perform read and write operations
+    #   only on an ACTIVE table.
+    #
+    #   You can use the DescribeTable action to check the table status.
+    #
+    # So we wait for the table to be created before yielding to the test.
+    #
+    # Link:  https://boto3.readthedocs.io/en/stable/reference/services/dynamodb.html#DynamoDB.Client.create_table
+    for i in range(10):
```

Comment:
nit (aimed at me, I think I wrote this code originally): we could write `for _ in range(10)`, as we never use the variable `i` in the loop.